src/main/java/teammates/logic/core/InstructorsLogic.java

```diff
@@ -143,6 +143,15 @@ public boolean isEmailOfInstructorOfCourse(String instructorEmail, String course
         return instructorsDb.getInstructorForEmail(courseId, instructorEmail) != null;
     }
 
+    /**
+     * Returns whether at least one instructor is displayed to the students.
+     */
+    public boolean isNoInstructorDisplayedToStudentsAfterEditing(String courseId, boolean isEditedInstructorDisplayed) {
+
+        List<InstructorAttributes> instructorsDisplayed = instructorsDb.getInstructorsDisplayedToStudents(courseId);
+        return (instructorsDisplayed.isEmpty()) || (instructorsDisplayed.size() == 1 && !isEditedInstructorDisplayed);
```

Comment:
Small logic gap (although unlikely to happen): if `instructorsDisplayed.isEmpty()` is true but `isEditedInstructorDisplayed` is also true, then there is still an instructor being displayed.

Reply:
So this scenario arises when, if you start a new session of teammates and navigate to the instructor course edit page. There all instructors have `isDisplayedToStudents` set to false (`instructorsDisplayed.isEmpty()` is true) and you check on one of the instructors to be displayed ('isEditedInstructorDisplayed' is true) but an exception is thrown. 

* I added the check `instructorsDisplayed.isEmpty()` as a part of defensive programming
* The instructor parameters are always updated after the checks have taken place, so `isEditedInstructorDisplayed` represents the new value.

Reply:
Bigger logic gap:
`instructorsDisplayed.size() == 1` does not imply that the sole displayed instructor is the one being edited. This conditional means that none of the other non-displayed instructors would be able to be edited.

Reply:
`instructorsDisplayed.size() == 1 && !isEditedInstructorDisplayed` means there is only instructor that is being displayed initially, the second part, `&& !isEditedInstructorDisplayed` ensures that the instructor that was edited , had its `isDisplayedToStudents` value changed from true to false. Hence if both parts are true it can be ascertained that the instructor that is being edited , was initially displayed to students, but now is being changed to not display. And since there were no other instructors with `isDisplayedToStudents` set to true, the exception must be thrown

Reply:
The main idea is that, the exception will be thrown when there is only one instructor that is being displayed to the students (given by `instructorsDisplayed.size() == 1`) and that particular instructor is being edited i.e it's value is changed from true to false (given by  `&& !isEditedInstructorDisplayed`). 

Consider a situation where there are X number of instructors. Out these, there is an instructor Y that is displayed and rest are not displayed i.e their `isDisplayedToStudents` is false. On editing, any of these instructors their `isDisplayedToStudents` value will change from false to true. While editing the instructor Y which is initially displayed, will change from true to false. This new value of false is given by `!isEditedInstructorDisplayed`.

Reply:
Is there any particular workflow that you find a potential bug in?

Reply:
I have successfully tested the possible workflows on my branch, maybe you could test it and let me know your concerns?